src/pages/workspace/accounting/AccountingContext.tsx

```diff
@@ -0,0 +1,139 @@
+import type {View} from 'react-native';
+import AccountingConnectionConfirmationModal from '@components/AccountingConnectionConfirmationModal';
+import useLocalize from '@hooks/useLocalize';
+import {removePolicyConnection} from '@libs/actions/connections';
+import Navigation from '@libs/Navigation/Navigation';
+import {isControlPolicy} from '@libs/PolicyUtils';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {ConnectionName} from '@src/types/onyx/Policy';
+import type Policy from '@src/types/onyx/Policy';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import {getAccountingIntegrationData} from './utils';
+
+type ActiveIntegration = {
+    name: ConnectionName;
+    shouldDisconnectIntegrationBeforeConnecting?: boolean;
+    integrationToDisconnect?: ConnectionName;
+};
+
+type ActiveIntegrationState = ActiveIntegration & {key: number};
+
+type AccountingContextType = {
+    activeIntegration?: ActiveIntegration;
+    startIntegrationFlow: (activeIntegration: ActiveIntegration) => void;
+
+    /*
+     * This stores refs to integration buttons, so the PopoverMenu can be positioned correctly
+     */
+    integrationRefs: RefObject<Record<string, MutableRefObject<View | null>>>;
+};
+
+const integrationRefsInitialValue = Object.values(CONST.POLICY.CONNECTIONS.NAME).reduce((acc, key) => {
+    acc[key] = {current: null};
+    return acc;
+}, {} as Record<ConnectionName, MutableRefObject<View | null>>);
+
+const defaultAccountingContext = {
+    activeIntegration: undefined,
+    startIntegrationFlow: () => {},
+    integrationRefs: {
+        current: integrationRefsInitialValue,
+    },
+};
+
+const AccountingContext = React.createContext<AccountingContextType>(defaultAccountingContext);
+
+type AccountingContextProviderProps = ChildrenProps & {
+    policy: Policy;
+};
+
+function AccountingContextProvider({children, policy}: AccountingContextProviderProps) {
+    const integrationRefs = useRef<Record<string, MutableRefObject<View | null>>>(defaultAccountingContext.integrationRefs.current);
+    const [activeIntegration, setActiveIntegration] = useState<ActiveIntegrationState>();
+    const {translate} = useLocalize();
+    const policyID = policy.id;
+
+    const startIntegrationFlow = React.useCallback(
+        (newActiveIntegration: ActiveIntegration) => {
+            const accountingIntegrationData = getAccountingIntegrationData(newActiveIntegration.name, policyID, translate);
+            if (accountingIntegrationData?.requiresControlPolicy && !isControlPolicy(policy)) {
+                const alias =
+                    newActiveIntegration.name === CONST.POLICY.CONNECTIONS.NAME.NETSUITE
+                        ? CONST.UPGRADE_FEATURE_INTRO_MAPPING.netsuite.alias
+                        : CONST.UPGRADE_FEATURE_INTRO_MAPPING.intacct.alias;
+                Navigation.navigate(ROUTES.WORKSPACE_UPGRADE.getRoute(policyID, alias));
+                return;
+            }
```

Comment:
We should configure it in the `accountingIntegrationData`. For example, we can add the `alias` property instead of the `requiresControlPolicy`.

Reply:
cc @war-in